packages/gallery/src/components/item/videos/videoItem.js

```diff
@@ -81,13 +97,60 @@ class VideoItem extends React.Component {
     return this.isHLSVideo() && utils.isiOS();
   }
 
+  updatePlayer = async () => {
+    if (!this.isYoutubeVideo()) {
+      return;
+    }
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      // if the user click play on iframe and state not yet updated break
+      if (this.state.shouldPlay && this.playingFromIframeClick) {
+        this.playingFromIframeClick = false;
+      }
+      if (!this.state.shouldPlay && this.pausingFromIframeClick) {
+        this.pausingFromIframeClick = false;
+      }
+      if (!this.state.shouldPlay && this.playingFromIframeClick) {
+        break;
+      }
+      if (this.state.shouldPlay && this.pausingFromIframeClick) {
+        break;
+      }
+      if (this.video?.player?.player?.player?.playVideo) {
+        const player = this.video.player.player.player;
+        const playerIsPlaying =
+          player.getPlayerState() === YOUTUBE_PLAY_STATES.PLAYING;
+
+        if (this.state.shouldPlay === playerIsPlaying) {
+          break;
+        }
+        if (this.state.shouldPlay && !this.pausingFromIframeClick) {
+          player.playVideo();
+        } else if (!this.state.shouldPlay && !this.playingFromIframeClick) {
+          player.pauseVideo();
+        }
+      }
+
+      // sleep until player is loaded
+      await utils.sleep(100);
+    }
+  };
+
   UNSAFE_componentWillReceiveProps(nextProps) {
     if (nextProps.shouldPlay || nextProps.firstUserInteractionExecuted) {
       this.setState({ loadVideo: true });
     }
 
-    if (nextProps.shouldPlay) {
-      this.setState({ shouldPlay: true });
+    if (nextProps.shouldPlay && !this.state.shouldPlay) {
+      this.playingFromIframeClick = false;
```

Comment:
You shouldn't alter `this`. Use only `props` and state for the behaviour of your component.